x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DataFrameAnalyticsManager.java

```diff
@@ -126,6 +126,9 @@ private void reindexDataframeAndStartAnalysis(DataFrameAnalyticsTask task, DataF
         // Reindexing is complete; start analytics
         ActionListener<RefreshResponse> refreshListener = ActionListener.wrap(
             refreshResponse -> {
+                if (task.isStopping()) {
+                    return;
```

Comment:
might be helpful for debugging to add a `debug` log here. I imagine it would be useful to figure out race conditions/CI failures.